plugin-build/plugin/src/main/java/io/sentry/kotlin/multiplatform/gradle/plugin/SentryPlugin.kt

```diff
@@ -0,0 +1,231 @@
+import org.jetbrains.kotlin.gradle.plugin.cocoapods.CocoapodsExtension
+import org.jetbrains.kotlin.gradle.plugin.cocoapods.KotlinCocoapodsPlugin
+import org.jetbrains.kotlin.gradle.plugin.mpp.Framework
+import org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget
+import org.jetbrains.kotlin.gradle.plugin.mpp.TestExecutable
+import java.io.ByteArrayOutputStream
+import java.io.File
+
+internal const val SENTRY_EXTENSION_NAME = "sentry"
+internal const val LINKER_EXTENSION_NAME = "linker"
+internal const val AUTO_INSTALL_EXTENSION_NAME = "autoInstall"
+internal const val COCOAPODS_AUTO_INSTALL_EXTENSION_NAME = "cocoapods"
+internal const val COMMON_MAIN_AUTO_INSTALL_EXTENSION_NAME = "commonMain"
+internal const val KOTLIN_EXTENSION_NAME = "kotlin"
+
+@Suppress("unused")
+class SentryPlugin : Plugin<Project> {
+    override fun apply(project: Project): Unit =
+        with(project) {
+            val sentryExtension =
+                project.extensions.create(
+                    SENTRY_EXTENSION_NAME,
+                    SentryExtension::class.java,
+                    project,
+                )
+            project.extensions.add(LINKER_EXTENSION_NAME, sentryExtension.linker)
+            project.extensions.add(AUTO_INSTALL_EXTENSION_NAME, sentryExtension.autoInstall)
+            project.extensions.add(
+                COCOAPODS_AUTO_INSTALL_EXTENSION_NAME,
+                sentryExtension.autoInstall.cocoapods,
+            )
+            project.extensions.add(
+                COMMON_MAIN_AUTO_INSTALL_EXTENSION_NAME,
+                sentryExtension.autoInstall.commonMain,
+            )
+
+            afterEvaluate {
+                val hasCocoapodsPlugin =
+                    project.plugins.findPlugin(KotlinCocoapodsPlugin::class.java) != null
+
+                if (sentryExtension.autoInstall.enabled.get()) {
+                    if (sentryExtension.autoInstall.commonMain.enabled.get()) {
+                        installSentryForKmp(sentryExtension.autoInstall.commonMain)
+                    }
+                    if (hasCocoapodsPlugin && sentryExtension.autoInstall.cocoapods.enabled.get()) {
+                        installSentryForCocoapods(sentryExtension.autoInstall.cocoapods)
+                    }
+                }
+
+                // If the user is not using the cocoapods plugin, linking to the framework is not automatic
+                // so we have to take care of that
+                if (!hasCocoapodsPlugin) {
+                    configureLinkingOptions(sentryExtension.linker)
+                }
+            }
+        }
+}
+
+internal fun Project.installSentryForKmp(commonMainAutoInstallExtension: SourceSetAutoInstallExtension) {
+    val kmpExtension = extensions.findByName(KOTLIN_EXTENSION_NAME)
+    if (kmpExtension !is KotlinMultiplatformExtension) {
+        // todo: log, not multiplatform found
+        return
+    }
+
+    val unsupportedTargets = listOf("wasm", "js", "mingw", "linux")
```

Comment:
there's also androidNative which I guess we also don't support through the KMP sdk, right? https://kotlinlang.org/docs/native-target-support.html#tier-3